iroh-net/src/magicsock/endpoint.rs

```diff
@@ -203,36 +201,40 @@ impl Endpoint {
             debug!("in `DEV_DERP_ONLY` mode, giving the DERP address as the only viable address for this endpoint");
             return (None, self.derp_region(), false);
         }
-        match self.best_addr {
-            Some(ref best_addr) => {
-                if !self.is_best_addr_valid(*now) {
-                    // We had a best_addr but it expired so send both to it and DERP.
-                    trace!(addr = %best_addr.addr, latency = ?best_addr.latency, "best_addr is set but outdated, use best_addr and derp");
-                    (Some(best_addr.addr), self.derp_region(), true)
-                } else {
-                    trace!(addr = %best_addr.addr, latency = ?best_addr.latency, "best_addr is set and valid, use best_addr only");
-                    // Address is current and can be used
-                    (Some(best_addr.addr), None, false)
-                }
+        // Update our best addr from candidate addresses (only if it is empty and if we have recent
+        // pongs).
+        self.update_best_addr_from_candidates();
+        match self.best_addr.state(*now) {
+            // we have a valid address: use it!
+            best_addr::State::Valid(best_addr) => {
+                trace!(addr = %best_addr.addr, latency = ?best_addr.latency, "best_addr is set and valid, use best_addr only");
```

Comment:
can you add `self.public_key.fmt_short()` here to know what peer we are working on?

Reply:
These log lines are only called from function that have instrumentation spans over them. So the log lines should look like this:
```
2023-10-23T09:20:32.398115Z TRACE poll_send{me=pp6ebwzioaeldcnl}:get_send_addrs{peer=cx6rezjzqm3ayple}: iroh_net::magicsock::endpoint: best_addr is set and valid, use best_addr only addr=172.20.0.1:60955 latency=Some(17.247288ms)
```